providers.tf.example

```diff
@@ -20,8 +21,12 @@ provider "helm" {
   kubernetes {
     host                   = data.aws_eks_cluster.cluster.endpoint
     cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
-    token                  = data.aws_eks_cluster_auth.cluster.token
+    #token                  = data.aws_eks_cluster_auth.cluster.token
+    exec {
```

Comment:
I am wondering why this problem occured only now. Did you try to update the AWS provider version? Maybe there is an incompatibility with the new k8s api version.

I would like to prevent using the exec command. If there is no better alternative this should be tested on Windows at least.

Reply:
ok, I can check that (updating provider) - the list of all provider versions is provided already (for comparison).  
this was run from WSL, maybe it has slightly different behaviour on Win.  
but in any case, there should be a possibility to run this from multiple platforms and getting same result.  
and this was tested using SSO, while ref arch docs mention using env vars (profile, region, token), so this might be an issue as well.

Reply:
Yes, you are right. It should work on Windows and Linux. Until now I did not observe any problems on Linux.

I had not problems with this aws profile for SSO (~/.aws/config)
```
[profile instance9]
region = eu-central-1
sso_account_id = 1234567890
sso_role_name = AdministratorAccess
sso_start_url = https://....awsapps.com/start#
sso_region = eu-central-1
output = json
```

I know that there are some [incompatibilities between different AWS Profile format versions and terraform](https://github.com/hashicorp/terraform/issues/32465). That is the reason why one should not use `aws configure sso --profile "profile"`

Reply:
Ran a few tests, region is required, regardless of setting used.  
On Win the 'exec' command is not required, but works just fine if both 'token' and 'exec' command are defined.  
I've removed the region and profile from vars, but it makes deployment slightly less elegant.

Reply:
I am trying to reproduce this error on linux. Can you send the link to the TF docs you mentioned above?

> added exec command to helm provider, to get auth token, as per TF docs

Reply:
here you go: https://registry.terraform.io/providers/hashicorp/helm/latest/docs#authentication

Reply:
btw, which TF version are you using?

Reply:
I am using TF v1.2.9